scripts-python/case.build

```diff
@@ -0,0 +1,508 @@
+def check_input_data(case):
+###############################################################################
+    run_cmd("./check_input_data --download")
+
+    get_refcase  = case.get_value("GET_REFCASE")
+    run_type     = case.get_value("RUN_TYPE")
+    continue_run = case.get_value("CONTINUE_RUN")
+
+    # JGF - I don't have the foggiest idea what this is for so this is just
+    # a 1-1 perl->python conversion
+    if (get_refcase == "TRUE" and run_type != "startup" and continue_run == "FALSE"):
+        din_loc_root = case.get_value("DIN_LOC_ROOT")
+        run_refdate  = case.get_value("RUN_REFDATE")
+        run_refcase  = case.get_value("RUN_REFCASE")
+        run_refdir   = case.get_value("RUN_REFDIR")
+        rundir       = case.get_value("RUNDIR")
+
+        refdir = os.path.join(run_refdir, run_refcase, run_refdate)
+        expect(os.path.isdir(refdir),
+"""
+*****************************************************************
+ccsm_prestage ERROR: $refdir is not on local disk
+obtain this data from the svn input data repository
+> mkdir -p %s
+> cd %s
+> cd ..
+> svn export --force https://svn-ccsm-inputdata.cgd.ucar.edu/trunk/inputdata/%s
+or set GET_REFCASE to FALSE in env_run.xml
+and prestage the restart data to $RUNDIR manually
+*****************************************************************""" % (refdir, refdir, refdir))
+
+        logging.info(" - Prestaging REFCASE (%s) to %s" % (refdir, rundir))
+
+        # prestage the reference case's files.
+
+        if (not os.path.exists(rundir)):
+            os.makedirs(rundir)
+
+        refcasefiles = glob.glob("%s/%s/*%s*" % (din_loc_root, refdir, run_refcase))
+        for rcfile in refcasefiles:
+            rcbaseline = os.path.basename(rcfile)
+            if (not os.path.exists("%s/%s" % (rundir, rcbaseline))):
+                os.symlink(rcfile, "%s/%s" % ((rundir, rcbaseline)))
+
+            # copy the refcases' rpointer files to the run directory
+            rpointerfiles = glob.glob("%s/%s/*rpointer*" % (din_loc_root, refdir))
+            for rpointerfile in rpointerfiles:
+                shutil.copy(rpointerfile, rundir)
+
+            cam2_list = glob.glob("%s/*.cam2.*" % rundir)
+            for cam2file in cam2_list:
+                camfile = cam2file.replace("cam2", "cam")
+                os.symlink(cam2file, camfile)
+
+            allrundirfiles = glob.glob("%s/*" % rundir)
+            for runfile in allrundirfiles:
+                os.chmod(runfile, 0755)
+
+###############################################################################
+def build_checks(case, build_threaded, comp_interface, use_esmf_lib, mpilib,
+                 nthrds_cpl, nthrds_atm, nthrds_lnd, nthrds_ice, nthrds_ocn, nthrds_glc, nthrds_wav, nthrds_rof, ninst_build, smp_value):
+###############################################################################
+    ninst_atm    = int(case.get_value("NINST_ATM"))
```

Comment:
I would really like to see all these things in an array or dict that we can loop over components, but that is a future cleanup.